xdsl/dialects/builtin.py

```diff
@@ -2159,6 +2221,18 @@ def create_dense_float(
 
         return DenseIntOrFPElementsAttr([type, BytesAttr(type.element_type.pack(data))])
 
+    @staticmethod
+    def create_dense_complex(
+        data_type: RankedStructure[ComplexType], data: Sequence[complex]
+    ) -> DenseIntOrFPElementsAttr[ComplexType]:
+        import itertools
+
```

Comment:
I think we should check that `data_type.element_type.element_type` is a float type here?